src/core/common/message.cpp

```diff
@@ -525,6 +525,48 @@ otError Message::Read(uint16_t aOffset, void *aBuf, uint16_t aLength) const
     return (ReadBytes(aOffset, aBuf, aLength) == aLength) ? OT_ERROR_NONE : OT_ERROR_PARSE;
 }
 
+bool Message::CompareBytes(uint16_t aOffset, const void *aBuf, uint16_t aLength) const
```

Comment:
```suggestion
bool Message::CompareBytes(uint16_t aOffset, const uint8_t *aBuf, uint16_t aLength) const
```

since the method name is `CompareBytes`.

Reply:
We intentionally use `void *` to make it easier for callers of these methods to be able to pass any pointer (instead of caller having to cast it, there is a common casting in the implementation). 

This is common pattern, we already use it in many places: for example, `ReadBytes()`, `AppendBytes()`, `WriteBytes()` (in `Message`) and `AppendTlv()`, `ReadTlv()` ... (in `Tlv`) or in `Sha256::Update()`, ...